### PR TITLE
Integrations: upstage LLM and Embeddings

### LINKED ISSUE #13192: [Feature Request]: add upstage llm and embeddings
### Feature Description

[Upstage](https://www.upstage.ai/solar-llm) has an awesome Solar LLM and Embeddings api. I would like to integrate this awesome API into llama index.

### Reason

Upstage's Solar is one of the most popular LLMs on the rise, and many Solar users are looking forward to integration with llama index.

### Value of Feature

_No response_

### PR DESCRIPTION
**Pull request by JuHyung-Son**
# Description

Upstage is one of the most up-and-coming LLM providers, and many of users want to use their LLM and Embeddings api, so in this PR, we've added an Upstage LLM and  Embeddings integration.

Also we deprecated and added warning on solar LLM integration which will not be managed anymore.

Fixes #13192 (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-upstage/README.md
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-upstage/README.md:1]: This README will show up on https://llamahub.ai -- you probably want to take some time to fill this out a bit more (for both the LLM and embeddings)

File: llama-index-integrations/embeddings/llama-index-embeddings-upstage/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-upstage/poetry.lock:1]: no need to commit the lock file, we just keep it for the top-level and core

File: llama-index-integrations/llms/llama-index-llms-upstage/README.md
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-upstage/README.md:1]: same comment on the readme

File: llama-index-integrations/llms/llama-index-llms-upstage/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-upstage/poetry.lock:1]: no need to commit this lock file

**Review by logan-markewich — COMMENTED** [review]
This looks good to me, mostly just a few nits

File: llama-index-integrations/embeddings/llama-index-embeddings-upstage/llama_index/embeddings/upstage/base.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        return {
+            "api_key": self.api_key,
+            "base_url": self.api_base,
+            "max_retries": self.max_retries,
+            "timeout": self.timeout,
+            "default_headers": self.default_headers,
+            "http_client": self._http_client,
+        }
+
+    def _get_query_embedding(self, query: str) -> List[float]:
```
[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply llama-index-integrations/embeddings/llama-index-embeddings-upstage/llama_index/embeddings/upstage/base.py:131]: Our eventual goal is to inherit OpenAIEmbedding perfectly as you said, but we still have some differences from actual api server.

**Comment by JuHyung-Son** [comment]
Thanks for detailed instruction.

**Review by logan-markewich — APPROVED** [review]

**Comment by JuHyung-Son** [comment]
@logan-markewich tests failed. But when I test it it success. do i need something to do additionally?

```
✕ llama-index-integrations/embeddings/llama-index-embeddings-upstage/tests/integration_tests/test_integrations.py:tests failed in 6.16s.
✕ llama-index-integrations/embeddings/llama-index-embeddings-upstage/tests/unit_tests/test_embeddings_upstage.py:tests failed in 6.48s.
```

**Comment by JuHyung-Son** [comment]
@logan-markewich Would you check this action?, 3.10 passed but 3.8 and 3.9 were canceled because of long running tasks.

**Comment by logan-markewich** [comment]
Can you merge main again? There was some issues that were recently fixed

**Comment by JuHyung-Son** [comment]
@logan-markewich action unit test failed with
```
==================================== ERRORS ====================================
_________ ERROR collecting tests/unit_tests/test_embeddings_upstage.py _________
ImportError while importing test module 'llama-index-integrations/embeddings/llama-index-embeddings-upstage/tests/unit_tests/test_embeddings_upstage.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
llama-index-integrations/embeddings/llama-index-embeddings-upstage/tests/unit_tests/test_embeddings_upstage.py:4: in <module>
    from llama_index.embeddings.upstage import UpstageEmbedding
E   ModuleNotFoundError: No module named 'llama_index.embeddings.upstage'
------------------------------- Captured stderr --------------------------------
```

I think I have pyproject set up correctly, all tests pass on my local as well, is there anything else I need to set up?

**Comment by logan-markewich** [comment]
@JuHyung-Son For the life of me I can't figure out why the embedding tests are failing. I've noticed that any class that inherits openai embeddings, pants (our cicd tool) does not properly detect dependencies.

My advice is we can keep the tests (since they are nice for local dev), but we need to wrap the import in a try/except in the tests, and skip if not found

**Comment by logan-markewich** [comment]
Also just and fyi, the embeddings pyproject.toml is missing the `llama-index-embeddings-openai` dependency (adding it doesn't fix tests, but is needed to publish the package properly)

**Pull request merged**